packages/strapi-generate-api/templates/bookshelf/model.template

```diff
@@ -5,64 +5,43 @@
  */
 
 module.exports = {
-
   // Before saving a value.
   // Fired before an `insert` or `update` query.
-  // beforeSave: function (model, attrs, options) {
-  //   return new Promise();
```

Comment:
Why are you removing this line? It's our way to explain that you have to return a Promise to make it work. The behaviour is different between Mongoose and Bookshelf.

Reply:
I forgot to notice that you added the `async` word to make sure these functions will return promises. But I remember that Bookshelf is using Bluebird promises which are not compatible with the native Node.js implementation. Maybe I'm wrong I just want to be sure that we're not introducing a bug here...

Reply:
In fact, returning a Promise or using `async/await` is dealing with the same thing.
Because if it returns a `new Promise()`, it need to handle with the `then()/catch()` block or `await` and we can reverse it.

For example, when I use Sequelize, a lot of functions return a Promise and I use it with the `async/await` logic without problem.

After, I based myself on the docs which indicated the use of `async/await` for Bookshelf as Mongoose 🙂

Reply:
@Aurelsicoko: I just see you add a second comment.

Normally Bluebird is not a problem to use it with `async/await`.
(and it seems that bluebird is not use actually)

Reply:
I need to test it before merging the PR. I remember that I already tried to make these functions async but it wasn't working as expected.